src/pixmap.c

```diff
@@ -300,6 +301,70 @@ void pixmap_free(struct pixmap* pm)
     free(pm->data);
 }
 
+bool pixmap_save(struct pixmap* pm, const char* path)
+{
+    FILE* fp;
+    unsigned i;
+
+    if (!(fp = fopen(path, "wb"))) {
+        return false;
+    }
+
+    // TODO: add alpha channel
+    fprintf(fp, "P6\n%zu %zu\n255\n", pm->width, pm->height);
+    for (i = 0; i < pm->width * pm->height; ++i) {
+        uint8_t color[4] = { (((pm->data[i] >> (8 * 2)) & 0xff)),
+                             (((pm->data[i] >> (8 * 1)) & 0xff)),
+                             (((pm->data[i] >> (8 * 0)) & 0xff)) };
+        fwrite(color, 3, 1, fp);
+    }
+
+    fclose(fp);
+    return true;
+}
+
+bool pixmap_load(struct pixmap* pm, const char* path)
+{
+    FILE* fp;
+    unsigned i;
+
+    if (!(fp = fopen(path, "rb"))) {
+        return false;
+    }
+
+    char header[3];
+    if (fscanf(fp, "%2s\n%zu %zu\n255\n", header, &pm->width, &pm->height) !=
+        3) {
+        goto fail;
+    }
+
+    if (strcmp(header, "P6")) {
```

Comment:
Just curious, what does "P6" mean?

Reply:
No, [it's not](https://github.com/artemsen/swayimg/blob/master/src/formats/bmp.c).
But you can add real BMP encoder for that =).

Reply:
Ah sorry, my mistake. It's PPM format. Probably the simplest thing out there.